### PR TITLE
First task-level preset with `BertClassifier`

### PR DESCRIPTION
**Pull request by jbischof**
Establish pattern for task-level presets using `bert_tiny_uncased_en_sst2`. This toy model will also be great for demos.

### CODE REVIEW

File: keras_nlp/models/bert/bert_presets_test.py [hunk 1]
```diff
@@ -184,7 +184,7 @@ def test_load_bert(self, load_weights):
     def test_load_bert_classifier(self, load_weights):
         for preset in BertClassifier.presets:
             classifier = BertClassifier.from_preset(
-                preset, num_classes=4, load_weights=load_weights
```
Comment (jbischof) [inline keras_nlp/models/bert/bert_presets_test.py:187]: Filed #493 for this bug.

File: keras_nlp/models/bert/bert_tasks.py [hunk 2]
```diff
@@ -110,11 +113,17 @@ def from_config(cls, config):
...
         return copy.deepcopy(backbone_presets)
 
+    @classproperty
+    def presets(cls):
+        return copy.deepcopy(backbone_presets | classifier_presets)
+
     @classmethod
-    @format_docstring(names=", ".join(backbone_presets))
+    @format_docstring(
+        names=", ".join(bert_presets.backbone_presets | classifier_presets)
```
Comment (jbischof) [inline keras_nlp/models/bert/bert_tasks.py:125]: @mattdangerw there is some crazy binding issue where your decorator aliases `backbone_presets` with the `classmethod` but also won't let me use it as a `dict`. Worth a discussion.

Comment (mattdangerw) [reply keras_nlp/models/bert/bert_tasks.py:125]: I would just pull this a constant above the class.

`PRESET_NAMES = ", ".join(bert_presets.backbone_presets.keys() + classifier_presets.keys())`

`@format_docstring(names=PRESET_NAMES)`

Would that work?

File: tools/checkpoint_training/bert_tiny_uncased_en_sst2_training.ipynb [hunk 3]
```diff
@@ -0,0 +1,4059 @@
...
+        }
+      ],
+      "source": [
+        "classifier = keras_nlp.models.BertClassifier.from_preset(\n",
+        "    \"bert_tiny_uncased_en\",\n",
+        "    num_classes=2,\n",
+        ")\n",
+        "classifier.compile(\n",
+        "    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),\n",
+        "    optimizer=keras.optimizers.experimental.AdamW(5e-5),\n",
```
[the remark this answers is not in the document]
Comment (mattdangerw) [reply tools/checkpoint_training/bert_tiny_uncased_en_sst2_training.ipynb:448]: Also, the paper suggests doing "random restarts" for fine tuning due to instability. That is, run this a few times with the dataset shuffled and new head weights for each run. Grab the highest performing model based on validation performance.

That doesn't seem like it would be too difficult to add, should we try that out? We should try to make sure we are roughly ballpark the original BERT performance, at least for BERT base.

**Review by mattdangerw — CHANGES_REQUESTED** [review]
Code looks great! Left a few comments re how we are generating these weights, but totally fine if we want to punt some of that to a follow up.

File: keras_nlp/models/bert/bert_presets.py [hunk 4]
```diff
@@ -220,3 +220,34 @@
         "vocabulary_hash": "bb6ca9b42e790e5cd986bbb16444d0e0",
     },
 }
+
+classifier_presets = {
+    "bert_tiny_uncased_en_sst2": {
```
[the remark this answers is not in the document]
Comment (jbischof) [reply keras_nlp/models/bert/bert_presets.py:225]: Yes this is a toy! My goal is to
1. make a toy model that will load quickly in a demo and
2. establish a pattern for task presets so we can farm them out ASAP (happy to start filing tasks)

We are not looking for SoTA performance in a demo, we want people to be able to run some code and get a sense for how the library works.

File: keras_nlp/models/bert/bert_tasks.py (hunk 2 above)
Comment (jbischof) [reply keras_nlp/models/bert/bert_tasks.py:125]: Seems clean!

File: tools/checkpoint_training/bert_tiny_uncased_en_sst2_training.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (jbischof) [reply tools/checkpoint_training/bert_tiny_uncased_en_sst2_training.ipynb:448]: Interesting, will try the clipping. I ran the colab several times and got almost identical results, so the current setup is stable if not necessarily optimal. I wonder if we should record performance in the `Description` field so we can update the checkpoint if we manage to improve it.

[the remark this answers is not in the document]
Comment (jbischof) [reply tools/checkpoint_training/bert_tiny_uncased_en_sst2_training.ipynb:448]: `clipnorm=1` had no effect on performance. The validation statistics have only changed in the third decimal or so between runs even with this option. We could also try sweeping the learning rate, but I doubt it matters much for our demo colab.

[the remark this answers is not in the document]
Comment (mattdangerw) [reply tools/checkpoint_training/bert_tiny_uncased_en_sst2_training.ipynb:448]: Got it! Thanks for checking.

I wonder if either, clipping it only needed for the deeper BERT variants more prone to vanishing/exploding gradients, or clipping is totally unnecessary and was just replicated to fine-tuning out of caution.

File: keras_nlp/models/bert/bert_presets.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mattdangerw) [reply keras_nlp/models/bert/bert_presets.py:225]: sgtm!

File: keras_nlp/models/bert/bert_tasks.py (hunk 2 above)
Comment (jbischof) [reply keras_nlp/models/bert/bert_tasks.py:125]: It turns out that `dict.keys()` does not return a `list` but a `dictview` so the above does not work. Luckily what does work is even simpler:

```python
PRESET_NAMES = ", ".join(list(backbone_presets) + list(classifier_presets))
```

**Review by mattdangerw — APPROVED** [review]
LGTM!

File: tools/checkpoint_training/bert_tiny_uncased_en_sst2_training.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (jbischof) [reply tools/checkpoint_training/bert_tiny_uncased_en_sst2_training.ipynb:448]: Good question! Will keep this in mind for bigger variants.
